File: erlang/p001.erl
-module(p001).
-export([main/0]).

p001(N) ->
  lists:sum([X || X <- lists:seq(1,N-1), X rem 3 =:= 0 orelse X rem 5 =:= 0 ]).

main() ->
  io:fwrite("~.10B~n",[p001(1000)]).

File: cpp/p001.cpp
#include <iostream>

using namespace std;

int main(){
  int sum=0;
  for(int i=0; i<1000; i++){
    if(!(i%3)||!(i%5)){
      sum += i;
    }
  }
  cout << sum << endl;
  return 0;
}
